### ISSUE TITLE
LDAP integration

### ISSUE THREAD

**Issue by smnschneider** [elastic/integrations#4278]
It would be useful to create a integration for getting data via LDAP (e.g. Active Directory)

There are a lot of security cases, where you need to get data from LDAP and compare those with your actual data.
It is also useful to track changes within groups or for specific user in more detail then getting it from Windows Event Logs.

So functions to get results for specific LDAP queries would be great.

**Comment by legoguy1000** [comment]
You're looking for a new input to query LDAP and index the results? Thats a interesting idea.

**Comment by smnschneider** [comment]
Yes that`s right. For the moment i use a Python script to query and ingest this data. A integration would help a lot here.

**Comment by jamiehynds** [comment]
@smnschneider integrating with directory services to ingest user and asset information is certainly on our roadmap. We are planning a proof of concept with AzureAD in the short-term to identify any architecture gaps we may have to overcome in order to ingest this data.

**Comment by willemdh** [comment]
Any news about this? Plus, please support ldaps.. :)

**Comment by jamiehynds** [comment]
@willemdh we are making good progress with an 'AzureAD Entity Analytics' package which will ingest user metadata. We plan on adding support for device metadata in the future too. 

The package is still being worked on, but you can view the PR here: https://github.com/elastic/integrations/pull/5638

**Comment by marc-gr** [comment]
> You're looking for a new input to query LDAP and index the results? Thats a interesting idea.

@jamiehynds do you think would be worth moving this into its own issue in beats?

**Comment by andrewkroh** [comment]
>  It is also useful to track changes within groups or for specific user in more detail then getting it from Windows Event Logs.

It sounds like this feature is met with the `entityanalytics_ad` package or associated Filebeat input (which connect to AD via LDAPS).

- https://www.elastic.co/docs/reference/integrations/entityanalytics_ad
- https://www.elastic.co/docs/reference/beats/filebeat/filebeat-input-entity-analytics#provider-activedirectory

If there's demand for generic LDAP queries, then please raise a new issue.
